### PR TITLE
Show wordpress icons

### PR DESCRIPTION
**Pull request by jjrohrer**
I wanted more help in tracking where my plugins were coming from.  I added a chunk that showed the wordpress icon if the plugin would be updated from wordpress.org.  So, now plugins get an icon if they are updated via one of the three major repositories, but no icon if they are hosted elsewhere.  I think this is a nice way to track/segregate the 'quasi-public' plugins from the 'quasi-private' plugins. 

BTW - I'm fairly new to the git community - so I'll apologize in advance for everything stupid that I've done, like, ever.

screenshot_101

### CODE REVIEW

**Comment by szepeviktor** [comment]
OK. Very nice.
I am used to sending PR-s but not receiving.
Please try to develop locally and keep 1 commit per PR.

**Comment by szepeviktor** [comment]
You could google for merging commits...

File: github-link.php [hunk 1]
```diff
@@ -9,7 +9,8 @@
 Author URI: http://www.online1.hu/webdesign/
 Domain Path:       /languages
 Text Domain:       github-link
-GitHub Plugin URI: https://github.com/szepeviktor/github-link
+GitHub Plugin URI: https://github.com/jjrohrer/github-link
```
Comment (szepeviktor) [inline github-link.php]: Please revert this.

File: README.md [hunk 2]
```diff
@@ -17,6 +17,12 @@ the GitHub icon is displayed **after** other plugin actions (links), otherwise i
 - `Bitbucket Plugin URI` shown as a Bitbucket logo ![Bitbucket logo](icon/bitbucket_32_darkblue_atlassian.png)
 - `Bitbucket Branch` shown as text after the Bitbucket icon
 
+
+### Wordpress.org headers
+
+- `Plugin URI`, if it contains the string 'wordpress.org', is shown as a Wordpress logo ![Wordpress logo](icon/wordpress_org_32x32.jpg)
```
Comment (szepeviktor) [inline README.md]: This icon shold stay for linking it from the README.

File: github-link.php [hunk 3]
```diff
@@ -9,7 +9,8 @@
...
+Depends: github-updater
```
Comment (szepeviktor) [inline github-link.php]: Please remove this line.
I don't use GHU. Don't tell anyone!!
Maybe others aren't using GHU...

File: github-link.php [hunk 4]
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
         }
     }
 
+
+    // if from wordpress.org, show that now.
+    if ( ! empty( $plugin_data['url']) && strstr($plugin_data['url'],'wordpress.org') !== FALSE) {
```
Comment (szepeviktor) [inline github-link.php]: WP coding style please:

``` php
false !== strstr( $plugin_data['url'], '//wordpress.org/plugins/' ) )
```

and split the two conditions into two lines as [above](https://github.com/szepeviktor/github-link/pull/3/files#diff-e570aa2cdf28400231de7f21099894b4L55).

File: github-link.php [hunk 5]
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
...
+        $icon = "icon/wordpress_org_32x32.jpg";
+        $new_action = array ('wordpress_org' => sprintf(
```
Comment (szepeviktor) [inline github-link.php:116]: Use a variable: `$wp_link_template` and copy $link_template's content. (title, link, target ...)

File: github-link.php [hunk 6]
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
...
+            '<img src="%s" style="vertical-align:-3px" height="16" width="16" alt="%s" />',
```
Comment (szepeviktor) [inline github-link.php]: Dashicons please. With color as inline style.

``` html
<span class="dashicons dashicons-wordpress"></span>
```

File: github-link.php [hunk 7]
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
...
+            plugins_url( $icon, __FILE__ ),
+            "A wordpress.org plugin."
```
Comment (szepeviktor) [inline github-link.php]: `__( "Visit WordPress.org Plugin Page" , "github-link" )`
And generate pot, po and mo.

File: github-link.php [hunk 8]
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
...
+        ) );
+        //$actions = $new_action + $actions;//ensure at front
```
Comment (szepeviktor) [inline github-link.php]: Only `// ensure WP icon at front`.

File: README.md (hunk 2 above)
Comment (szepeviktor) [inline README.md]: Sorry. Use https://s.w.org/about/images/logos/wordpress-logo-32-blue.png
from https://wordpress.org/about/logos/

File: github-link.php (hunk 1 above)
Comment (jjrohrer) [inline github-link.php]: Yike - my newness shows.  I thought a pull request would point to specific commit.  This was added later.  Reverting now.

File: README.md (hunk 2 above)
Comment (jjrohrer) [inline README.md]: I think this is OBE by the dashicon

File: github-link.php (hunk 4 above)
Comment (jjrohrer) [inline github-link.php]: done

File: github-link.php (hunk 5 above)
Comment (jjrohrer) [inline github-link.php:116]: Sorry - I'm not following what you are suggesting.

File: README.md (hunk 2 above)
Comment (szepeviktor) [inline README.md]: Needed for the readme.

File: github-link.php (hunk 5 above)
Comment (szepeviktor) [inline github-link.php:116]: Define a template like in `$link_template`.

File: github-link.php (hunk 7 above)
Comment (jjrohrer) [inline github-link.php]: I've never done that - I wouldn't even know where to start, but I replaced the previous text with __(...)

**Comment by szepeviktor** [comment]
There is a fundamental problem. PLugin authors put their own blog's URL in `Plugin URI`.
@afragen Could you help us to detect plugin origin **without an HTTP request**?

**Comment by szepeviktor** [comment]
@afragen Just tell me I already know how to: https://github.com/szepeviktor/github-link/blob/master/github-link.php#L55-L58

File: github-link.php [hunk 9]
```diff
@@ -10,6 +10,7 @@
 Domain Path:       /languages
 Text Domain:       github-link
 GitHub Plugin URI: https://github.com/szepeviktor/github-link
+Depends: GitHub Updater
```
Comment (szepeviktor) [inline github-link.php:13]: I'll finish this PR. Thank you!!

**Comment by afragen** [comment]
If you're trying to detect if the plugin is NOT using GHU, it's probably easier to check the `empty()` of `GitHub Plugin URI` and `Bitbucket Plugin URI` and then assume hosting on wp.org.

Also, `GitHub Access Token` header no longer exists.

File: github-link.php (hunk 3 above)
Comment (afragen) [inline github-link.php]: I won't tell anyone. :wink:

**Comment by jjrohrer** [comment]
I would specifically recommend against assuming wp.org http://wp.org/.  My own install, for example, is about 1/3 ‘other’.  They are privately hosted because they are commercial plugins or custom plugins still underdevelopment.  This is my original use-case for the WP icons - to find out which plugins where part of a central repository, vs which ones part of some other quasi-private repository.
JJ

**Comment by szepeviktor** [comment]
Thank you!

**Comment by szepeviktor** [comment]
As I mentioned WP.org detection was already in the code. https://github.com/szepeviktor/github-link/blob/master/github-link.php#L56-L60

**Comment by afragen** [comment]
That won't work. GHU adds to `$plugin_state->response[ $plugin_file ]` when an update is detected.

**Comment by szepeviktor** [comment]
Oh no! What do you suggest? (what is not a HTTP request)

**Comment by afragen** [comment]
If you take look at our last GHU commit re:View details. It seems that only .org plugins and now GHU plugins will show View details. Others will show 'Visit plugin site'. Some combination of checking his and having GitHub/Bitbucket Plugin URI empty could be a solution.

**Comment by szepeviktor** [comment]
How about searching for `//wordpress.org/plugins/` in `$plugin_state->response[ $plugin_file ]->url`?

**Comment by szepeviktor** [comment]
Noone should write:

```
GitHub Plugin URI: https://wordpress.org/plugins/post-tags-and-categories-for-pages/
```

or `https://my-site.net//wordpress.org/plugins/bingo`

**Comment by szepeviktor** [comment]
``` php
// GHU also sets plugin->url
if ( false !== strstr( $plugin_uri, '//wordpress.org/plugins/' ) ) {
    $new_action = array( 'wordpress_org' => sprintf(
        $wp_link_template,
        $plugin_uri,
        __( "Visit WordPress.org Plugin Page" , "github-link" )
    ) );
    $actions = array_merge( $new_action, $actions );
}
```

**Comment by afragen** [comment]
> How about searching for //downloads.wordpress.org/plugin/ in $plugin_state->response[ $plugin_file ]->url?

That should work. Don't forget to check $plugin_state->no_updates too.

**Comment by szepeviktor** [comment]
Thank you!
Done. https://github.com/szepeviktor/github-link/commit/5a8c226224854887f4b8161b971b479ce7af3a9a#diff-e570aa2cdf28400231de7f21099894b4R117

**Pull request closed without merging**